test/spinxsk/spinxsk.c

```diff
@@ -1992,6 +2025,97 @@ ProcessPkts(
     return TRUE;
 }
 
+#define FNMP_FRAME_MAX_BACKFILL 64
+#define FNMP_IPV4_ADDRESS "192.168.200.1"
+#define FNMP_NEIGHBOR_IPV4_ADDRESS "192.168.200.2"
+#define FNMP_IPV6_ADDRESS "fc00::200:1"
+#define FNMP_NEIGHBOR_IPV6_ADDRESS "fc00::200:2"
+#define FNMP_LOCAL_ETHERNET_ADDRESS_INIT {0x22, 0x22, 0x22, 0x22, 0x00, 0x00}
+#define FNMP_NEIGHBOR_ETHERNET_ADDRESS_INIT {0x22, 0x22, 0x22, 0x22, 0x00, 0x02}
+
+typedef union _INET_ADDR_STORAGE {
+    IN_ADDR Ipv4;
+    IN6_ADDR Ipv6;
+} INET_ADDR_STORAGE;
+
+//
+// Build and inject UDP-like frames in the receive path through the FNMP driver.
```

Comment:
we also need TCP-ish and QUIC-ish.

Reply:
Adding TCP.
What would be the difference between UDP-ish and QUIC-ish from the point of view of XDP?

Reply:
QUIC-ish has unique filtering rules, e.g. XDP_MATCH_QUIC_FLOW_DST_CID, and QUIC-ish might not be UDP in the case of QTIP! That has its own set of filtering rules, like XDP_MATCH_TCP_QUIC_FLOW_SRC_CID.

So the "-ish" suffix is doing some heavy lifting here :)